src/Quasar/Mount/Couchbase.purs

```diff
@@ -20,75 +20,84 @@ module Quasar.Mount.Couchbase
 import Data.Bifunctor (lmap)
+import Data.Codec (decode, encode)
 import Data.Either (Either(..))
-import Data.List as L
 import Data.Maybe (Maybe(..), maybe, fromMaybe)
 import Data.Newtype (un)
 import Data.Number as Num
-import Data.Path.Pathy ((</>))
-import Data.Path.Pathy as P
 import Data.StrMap as SM
+import Data.String.NonEmpty (NonEmptyString)
 import Data.Time.Duration (Seconds(..))
 import Data.Tuple (Tuple(..))
-import Data.URI as URI
-import Data.URI.AbsoluteURI as AbsoluteURI
-import Quasar.Mount.Common (Host) as Exports
-import Quasar.Mount.Common (Host, extractHost)
+import Data.URI.Scheme as Scheme
+import Pathy (Name(..), (</>))
+import Pathy as P
+import Quasar.Data.URI as URI
 
 type Config =
-  { host ∷ Host
-  , bucketName ∷ String
+  { host ∷ URI.QURIHost
+  , bucketName ∷ Maybe NonEmptyString
   , password ∷ String
   , docTypeKey ∷ String
   , queryTimeout ∷ Maybe Seconds
   }
 
 toJSON ∷ Config → Json
 toJSON config =
-  let uri = AbsoluteURI.print (toURI config)
+  let uri = encode URI.qAbsoluteURI (toURI config)
   in "couchbase" := ("connectionUri" := uri ~> jsonEmptyObject) ~> jsonEmptyObject
 
 fromJSON ∷ Json → Either String Config
 fromJSON
   = fromURI
-  <=< lmap show <<< AbsoluteURI.parse
+  <=< lmap show <<< decode URI.qAbsoluteURI
   <=< (_ .? "connectionUri")
   <=< (_ .? "couchbase")
   <=< decodeJson
 
-toURI ∷ Config → URI.AbsoluteURI
+toURI ∷ Config → URI.QAbsoluteURI
 toURI { host, bucketName, password, docTypeKey, queryTimeout } =
   URI.AbsoluteURI
-    (Just uriScheme)
-    (URI.HierarchicalPart
-      (Just (URI.Authority Nothing (pure host)))
-      (Just (Right (P.rootDir </> P.file bucketName))))
-    (Just (URI.Query props))
+    uriScheme
+    hierarchicalPart
+    (Just (URI.QueryPairs props))
   where
-  props ∷ L.List (Tuple String (Maybe String))
-  props = L.Nil
-    <> pure (Tuple "password" (Just password))
-    <> pure (Tuple "docTypeKey" (Just docTypeKey))
-    <> maybe L.Nil (pure <<< Tuple "queryTimeoutSeconds" <<< Just <<< show <<< un Seconds) queryTimeout
+  hierarchicalPart :: URI.QHierarchicalPart
+  hierarchicalPart =
+    URI.HierarchicalPartAuth
+      authority
+      (case bucketName of
+        Nothing -> Just $ Left P.rootDir
+        Just n -> Just $ Right $ P.rootDir </> P.file' (Name n)
+      )
 
-fromURI ∷ URI.AbsoluteURI → Either String Config
-fromURI (URI.AbsoluteURI scheme (URI.HierarchicalPart auth path) query) = do
-  unless (scheme == Just uriScheme) $ Left "Expected 'couchbase' URL scheme"
-  host ← extractHost auth
+  authority :: URI.QAuthority
+  authority = URI.Authority Nothing host
+
+  props ∷ Array (Tuple String (Maybe String))
+  props =
+    [ Tuple "password" (Just password)
+    , Tuple "docTypeKey" (Just docTypeKey)
+    ] <> maybe [] (pure <<< Tuple "queryTimeoutSeconds" <<< Just <<< show <<< un Seconds) queryTimeout
+
+fromURI ∷ URI.QAbsoluteURI → Either String Config
+fromURI (URI.AbsoluteURI scheme (URI.HierarchicalPartNoAuth path) query) = 
```

Comment:
Could switch this (and any other like it) to a `case _ of` while we're here perhaps?